network/dag/dag.go

```diff
@@ -146,7 +147,13 @@ func (d *dag) Migrate() error {
 				for _, ref := range heads {
 					transaction, err := getTransaction(ref, tx)
 					if err != nil {
+						if errors.Is(err, ErrTransactionNotFound) {
+							return fmt.Errorf("database migration failed: head not found (%s=%s)", core.LogFieldTransactionRef, ref)
```

Comment:
Again, the error message is lost. Can *never* do that. Either log, return or deal/absorb the error.

Reply:
This case should never happen. If this fails the node won't start and the error here provides a more useful debugging info than the returning the original error. (head is a specific transaction)

Reply:
now wraps the original error

Reply:
You can not make assumptions on how useful an error is. Never! The logic may change over time. All that is required is to preserve and amend with any decision making along the execution chain.